receiver/sqlserverreceiver/metadata.yaml

```diff
@@ -255,6 +263,145 @@ metrics:
+      aggregation_temporality: cumulative
+      monotonic: false
+      value_type: int
+    attributes: [table.state, table.status]
+  sqlserver.database.backup_or_restore.rate:
+    enabled: false
+    description: The number of backups/restores per second.
+    unit: “{backups_or_restores}/s”
+    gauge:
+      value_type: int
+    attributes: []
+  sqlserver.replica.sent.rate:
+    enabled: false
+    description: Bytes sent per second to a replica.
+    unit: “{bytes}/s”
+    gauge:
+      value_type: double
+    attributes: []
+  sqlserver.replica.received.rate:
+    enabled: false
+    description: Bytes received per second from a replica.
+    unit: “{bytes}/s”
+    gauge:
+      value_type: double
+    attributes: []
+  sqlserver.database.execution_errors.count:
+    enabled: false
+    description: Number of execution errors.
+    unit: “{errors}”
+    sum:
+      aggregation_temporality: cumulative
+      monotonic: false
+      value_type: int
+    attributes: []
+  sqlserver.memory.free_list_stalls.rate:
+    enabled: false
+    description: Number of free list stalls per second.
+    unit: “{stalls}/s”
+    gauge:
+      value_type: int
+    attributes: []
+  sqlserver.database.free_space_tempdb:
+    enabled: false
+    description: Total free space in temporary DB.
+    unit: “KB”
+    gauge:
+      value_type: int
+    attributes: []
+  sqlserver.database.full_scan.rate:
+    enabled: false
+    description: The number of unrestricted full table or index scans per second.
+    unit: "{scans}/s"
+    gauge:
+      value_type: int
+    attributes: []
+  sqlserver.index.search.rate:
+    enabled: false
+    description: Total number of index searches per second.
+    unit: “{searches}/s”
+    gauge:
+      value_type: double
+    attributes: []
+  sqlserver.database.lock_timeout.rate:
+    enabled: false
+    description: Total number of lock timeouts per second.
+    unit: “{lock_timeouts}/s”
+    gauge:
+      value_type: double
+    attributes: []
+  sqlserver.database.login.rate:
+    enabled: false
+    description: Total number of logins per second.
+    unit: “{logins}/s”
+    gauge:
+      value_type: double
+    attributes: []
+  sqlserver.database.logout.rate:
+    enabled: false
+    description: Total number of logouts per second.
+    unit: “{logouts}/s”
+    gauge:
+      value_type: double
+    attributes: []
+  sqlserver.database.deadlock.rate:
+    enabled: false
+    description: Total number of deadlocks per second.
+    unit: “{deadlocks}/s”
+    gauge:
+      value_type: double
+    attributes: []
+  sqlserver.database.mirror_write_transaction.rate:
+    enabled: false
+    description: Total number of mirror write transactions per second.
+    unit: “{mirror_write_transactions}/s”
+    gauge:
+      value_type: double
+    attributes: []
+  sqlserver.memory.grants_pending.count:
+    enabled: false
+    description: Total number of memory grants pending.
+    unit: “{memory_grants_pending}”
+    sum:
```

Comment:
Do you know what exactly this metric defines? Is it only showing current pending grants? Is it showing pending grants over the last x seconds? If it's the former, then I think how non-monotonic cumulative sum is fine. If it's the latter, then I think a gauge makes more sense.

Reply:
This metric states " Memory Grants Pending displays the total number of SQL Server processes that are waiting to be granted workspace in the memory."  And you can grab the value using the following SQL query.
```
SELECT object_name, counter_name, cntr_value
FROM sys.dm_os_performance_counters
WHERE [object_name] LIKE '%Memory Manager%'
AND [counter_name] = 'Memory Grants Pending'
```

To answer your question it is the former nor the latter and counter is needed. For your reference, further details can found succinctly [here](https://learn.microsoft.com/en-us/troubleshoot/sql/database-engine/performance/troubleshoot-memory-grant-issues)